pynest/nest/lib/hl_api_types.py

```diff
@@ -268,16 +267,22 @@ def __eq__(self, other):
 
         if self.__len__() != other.__len__():
             return False
+
         return sli_func('eq', self, other)
 
     def __neq__(self, other):
         if not isinstance(other, NodeCollection):
             raise NotImplementedError()
+
         return not self == other
 
     def __len__(self):
         return sli_func('size', self._datum)
 
+    def __int__(self):
+        # this will automatically raise an error if len is not 1
+        return int(self.get('global_id'))
```

Comment:
Is this function still needed?

Reply:
No, but since it was written, I left it... should I remove it?

Reply:
If it has nothing to do with the NumPy array conversion, it should be removed in this PR.